service/history/engine/engineimpl/poll_mutable_state.go

```diff
@@ -160,6 +160,15 @@ func (e *historyEngineImpl) getMutableStateOrLongPoll(
 		return nil, err
 	}
 
+	if response.GetVersionHistories() == nil {
```

Comment:
🤔 since `.GetX()` is generally a nil-hiding thing, tbh I think this might be worth changing to `response.VersionHistories`.

in practice this is fine:
```go
func (v *GetMutableStateResponse) GetVersionHistories() (o *VersionHistories) {
	if v != nil && v.VersionHistories != nil {
		return v.VersionHistories
	}
	return
}
```
but that's sorta abnormal/unexpected for those methods imo.  it'd make sense if it was a `**VersionHistories`, but it's just weird code otherwise (it amounts to `if v != nil { return v.VersionHistories }` but it's generated code so it's not simplified)